osgar/drivers/test_realsense.py

```diff
@@ -224,19 +225,33 @@ def test_depth(self):
         c = RealSense(bus=bus.handle('rs'), config={})
         tester = bus.handle('tester')
         bus.connect('rs.depth', 'tester.depth')
+        bus.connect('rs.color', 'tester.color')
         frameset = MagicMock()
         frameset.is_frameset.return_value = True
         frame = frameset.as_frameset.return_value.get_depth_frame.return_value
         frame.get_timestamp.return_value = 0
         frame.get_frame_number.return_value = 0
         frame.is_depth_frame.return_value = True
         frame.as_depth_frame.return_value.get_data.return_value = [1,2]
+
+        color_frame = frameset.as_frameset.return_value.get_color_frame.return_value
+        color_frame.get_timestamp.return_value = 0
+        color_frame.get_frame_number.return_value = 0
+        color_frame.is_frame.return_value = True
+        color_frame.as_frame.return_value.get_data.return_value = np.asarray([[0, 100, 255]], dtype=np.uint8)
+
         c.depth_callback(frameset)
-        dt, channel, depth = tester.listen()
+        dt, channel_1, depth = tester.listen()
+        dt, channel_2, color = tester.listen()
         depth_expected = np.asanyarray([1,2])
-        self.assertEqual(channel, 'depth')
+        color_expected = np.asanyarray([[0, 100, 255]], dtype=np.uint8)
+        self.assertEqual(channel_1, 'depth')
+        self.assertEqual(channel_2, 'color')
+        color = cv2.imdecode(np.fromstring(color, dtype=np.uint8), 0)
         self.assertEqual(depth.shape, depth_expected.shape)
         self.assertEqual(depth.dtype, depth_expected.dtype)
         self.assertTrue(np.array_equal(depth, depth_expected))
+        self.assertEqual(color.shape, color_expected.shape)
+#        self.assertTrue(np.array_equal(color, color_expected))
 
```

Comment:
Here i got unexpected result:

`self.assertTrue(np.array_equal(color, color_expected))`
`AssertionError: False is not true`

`print(color, color_expected)`
`[[  0  99 255]] [[  0 100 255]]`

Maybe jpeg compression?

Reply:
Well, you have
```python
color_frame.as_frame.return_value.get_data.return_value = np.asarray([[0, 100, 255]], dtype=np.uint8)
```
and you run jpeg decompression on it - which will not work since it is not jpeg compressed data - it is just plain numpy array with 3 values. Just skip the cv2.imdecode.